### PR TITLE
Introduce `cider-shorten-error-overlays` customization option

### LINKED ISSUE #3525: Make the error overlay trimming optional
## Context

https://github.com/clojure-emacs/cider/discussions/3338#discussioncomment-7286764

## Task 

Extract https://github.com/clojure-emacs/cider/blob/b83c997ec2473254f97e8754c217ce2d485b9d33/cider-eval.el#L860-L864 to a defun, and make its application optional.

### PR DESCRIPTION
**Pull request by vemv**
Fixes #3525

### CODE REVIEW

**Comment by bbatsov** [comment]
Seems to me we don't really need a defcustom for this. Given that the offending line would already be underlined this should simply be the only behavior. There's no need to make everything customizable. :-)

**Comment by bbatsov** [comment]
Hmm, it seems to me we're solving the wrong problem here - if the highlighting is not precise/useful probably we should improve it. Obviously adding this extra defcustom is not a big deal for us, but it just seems weird to me to have such a high granularity of customizations, that probably almost no one will ever touch.

**Comment by vemv** [comment]
My take would be that different Clojure developers need/want different degrees of detail, for errors.

Personally, I feel that I have seen certain errors so many times, that just a short string will suffice to inform me.

But not everyone operates like that.

**Comment by bbatsov** [comment]
> My take would be that different Clojure developers need/want different degrees of detail, for errors.

This part I get, but I believe that level of granularity is something like a line in an overlay vs a stacktrace buffer. Now we're discussing whether we need a defcustom for a message. Every such branch/defcustom adds some maintenance complexity, that's why it's good to consider carefully their value and their alternatives.

File: cider-eval.el [hunk 1]
```diff
@@ -821,6 +821,24 @@ REPL buffer.  This is controlled via
                                                      conn)))
           (nrepl-dict-get result "phase"))))))
 
+(defcustom cider-shorten-error-overlays t
+  "If t, overlays that reflect error messages will be more concise.
+Specifically, the prefix matched by `cider-clojure-compilation-regexp',
+and the suffix matched by `cider-module-info-regexp' will be removed."
+  :type 'boolean
+  :group 'cider
+  :package-version '(cider . "0.19.0"))
```
Comment (bbatsov) [inline cider-eval.el:830]: Btw, this should be 1.9, not 0.19. :-)

**Comment by vemv** [comment]
Maybe a middle ground solution would be to make `cider--shorten-error-message` public such that users can advice it to act like the identity function.

File: cider-eval.el [hunk 2]
```diff
@@ -821,6 +821,24 @@ REPL buffer.  This is controlled via
                                                      conn)))
           (nrepl-dict-get result "phase"))))))
 
+(defcustom cider-shorten-error-overlays t
```
Comment (bbatsov) [inline cider-eval.el:824]: I'd name this `cider-shorten-inline-errors` or `cider-shorten-inline-error-messages`. I think that the overlay is an implementation detail, that might change in the future, so it's better to focus on the essence (that we should an error message inline).

File: doc/modules/ROOT/pages/usage/code_evaluation.adoc [hunk 3]
```diff
@@ -207,6 +207,15 @@ bottom) with the `cider-use-overlays` variable:
 (setq cider-use-overlays nil)
 ----
 
+Overlays that indicate errors are by default trimmed of file/line/phase information.
+If you find it useful, you can customize:
```
Comment (bbatsov) [inline doc/modules/ROOT/pages/usage/code_evaluation.adoc:211]: Might be good to also put two error messages side by side here, so it clearer to people how the messages look in both cases.

**Pull request merged**